pkg/bff/gds.go

```diff
@@ -284,3 +286,82 @@ func (s *Server) Register(c *gin.Context) {
 
 	c.JSON(http.StatusOK, out)
 }
+
+// VerifyContact is currently a passthrough helper that forwards the verify contact
+// request from the user interface to the GDS that needs contact verification.
+func (s *Server) VerifyContact(c *gin.Context) {
+	// Bind the parameters associated with the verify contact request
+	params := &api.VerifyContactParams{}
+	if err := c.ShouldBindQuery(&params); err != nil {
+		log.Warn().Err(err).Msg("could not bind request with query params")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		return
+	}
+
+	// Ensure that we have all required parameters
+	if params.ID == "" || params.Token == "" || params.Directory == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("must provide vaspID, token, and registered_directory in query parameters"))
+		return
+	}
+
+	// Ensure the registered_directory is one we understand
+	params.Directory = strings.ToLower(params.Directory)
+	if params.Directory != trisatest && params.Directory != vaspdirectory {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("unknown registered directory"))
+	}
+
+	// Make the GDS request
+	log.Debug().Str("registered_directory", params.Directory).Msg("issuing GDS verify contact request")
+	req := &gds.VerifyContactRequest{Id: params.ID, Token: params.Token}
+	ctx, cancel := context.WithTimeout(c.Request.Context(), 25*time.Second)
+	defer cancel()
+
+	var (
+		err error
+		rep *gds.VerifyContactReply
+	)
+
+	switch params.Directory {
+	case trisatest:
+		rep, err = s.testnet.VerifyContact(ctx, req)
+	case vaspdirectory:
+		rep, err = s.mainnet.VerifyContact(ctx, req)
+	default:
+		log.Error().Str("registered_directory", params.Directory).Str("endpoint", "verify").Msg("unhandled directory")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not verify contact"))
+		return
+	}
+
+	// Handle GDS errors
+	if err != nil {
+		serr, _ := status.FromError(err)
+		switch serr.Code() {
+		case codes.InvalidArgument:
+			c.JSON(http.StatusBadRequest, api.ErrorResponse(serr.Message()))
+		case codes.NotFound:
+			c.JSON(http.StatusNotFound, api.ErrorResponse(serr.Message()))
+		case codes.Aborted:
+			c.JSON(http.StatusConflict, api.ErrorResponse(serr.Message()))
+		default:
+			log.Error().Err(err).Str("code", serr.Code().String()).Str("registered_directory", params.Directory).Msg("could not verify contact")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse(serr.Message()))
+		}
+		return
+	}
```

Comment:
I researched the errors that GDS `VerifyContact` will return; in addition to `InvalidArgument`, `NotFound`, and `Aborted`, it will also return `Internal` and `FailedPrecondition` - but these are both 500 errors. GDS returns a user-friendly error, so it is safe to just return the error message from the server.